turbopack/crates/turbopack-ecmascript/src/tree_shake/graph.rs

```diff
@@ -550,21 +559,36 @@ impl DepGraph {
             modules.push(chunk);
         }
 
-        exports.insert(Key::Exports, modules.len() as u32);
-
-        for star in &star_reexports {
+        if !star_reexports.is_empty() {
+            // `Exports` module should export everything from the re-exported modules
             exports_module
                 .body
-                .push(ModuleItem::ModuleDecl(ModuleDecl::ExportAll(star.clone())));
+                .push(ModuleItem::ModuleDecl(ModuleDecl::ExportAll(ExportAll {
+                    span: DUMMY_SP,
+                    src: Box::new(TURBOPACK_PART_IMPORT_SOURCE.into()),
+                    type_only: false,
+                    with: Some(Box::new(create_turbopack_part_id_assert(
+                        PartId::StarReexports,
+                    ))),
+                })));
+
+            for star in &star_reexports {
+                star_reexports_module
+                    .body
+                    .push(ModuleItem::ModuleDecl(ModuleDecl::ExportAll(star.clone())));
+            }
+
+            results.insert(Key::StarReexports, modules.len() as u32);
+            modules.push(star_reexports_module);
         }
```

Comment:
You don't want to generate source code for these reexporting modules. They can be created without source code.